VoodooI2C/VoodooI2C/VoodooI2CController/VoodooI2CACPIController.cpp

```diff
@@ -28,13 +28,13 @@ IOReturn VoodooI2CACPIController::setPowerState(unsigned long whichState, IOServ
     if (whatDevice != this)
         return kIOPMAckImplied;
 
-    if (whichState == kIOPMPowerOff) {
-        physical_device.awake = false;
-        unmapMemory();
-
-        setACPIPowerState(kVoodooI2CStateOff);
-
-        IOLog("%s::%s Going to sleep\n", getName(), physical_device.name);
+    if (whichState == 0) {  // index of kIOPMPowerOff state in VoodooI2CIOPMPowerStates
```

Comment:
Why are you not comparing the index directly? What if VoodooI2CIOPMPowerStates changes?

Reply:
`kIOPMPowerOff` is not an index but a flag like `kIOPMPowerOn` in `IOPMPowerFlags`.

> What if VoodooI2CIOPMPowerStates changes?

will have to change the condition correspondingly, because `whichState` here is only the ordinal number in the power state array of the state that the driver is being notified to switch to.